### DISCUSSION TITLE
`take` argument could accept `NaN` and use it as `undefined`

### DISCUSSION THREAD

**Discussion by paulosalvatore** [prisma/prisma#6099] [Q&A] [answered]
As title suggests, right now if `NaN` is given to `take` argument, it throws an error:

```bash
Argument take: Got invalid value NaN on prisma.findManyEventGroup. Provided Float, expected Int.
```

In order to fix that, we have to use the following syntax:

```js
take: numberVariable || undefined
```

Would be great if prisma did that by itself, but I'm not sure if you guys agree to that.

**Comment by ryands17** [comment, marked as answer]
@paulosalvatore 👋 

I think it would be better in this case to validate the input before passing it to Prisma with something like [yup](https://github.com/jquense/yup] or [zod](https://github.com/colinhacks/zod) so that you can send the validation back to the user on invalid input.

**Reply by paulosalvatore** [reply]
Alright, @ryands17.

Thanks for you feedback :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
